src/en/TravisTranslations.lua

```diff
@@ -0,0 +1,225 @@
+		if v:hasAttr("border") then
+			v:removeAttr("border")
+		end
+	end, nil, true))
+	for _,v in pairs(toRemove) do
+		v:remove()
+	end
+	return pageOfElem(chap, true)
+end
+
+local function parseNovel(novelURL, loadChapters)
+	local doc = GETDocument(expandURL(novelURL))
+	local content = doc:selectFirst("main#primary")
+
+	local info = NovelInfo {
+		title = content:selectFirst("h1#heading"):attr("title"),
+		imageURL = "https:" .. content:selectFirst("img"):attr("data-src"),
+		status = ({
+			Completed = NovelStatus.COMPLETED,
+			Ongoing = NovelStatus.PUBLISHING
+		})[content:selectFirst(".grid.grid-cols-2.gap-3.my-2"):child(3):selectFirst("strong"):text()],
+		description = table.concat(map(doc:selectFirst('div[property="description"]'):select("p"), text), '\n'),
+		artists = { "Translator: Travis Translations" },
+		genres = map(content:selectFirst("ul.space-x-1.space-y-2"):select("a"), text),
+		tags = map(content:selectFirst("ul.flex.flex-wrap.my-2"):select("a"), text)
+	}
+
+	if loadChapters then
+		local chapters = (map(content:selectFirst('div[x-show="tab === \'toc\'"]'):select("li"), function(v, i)
+			-- This is to ignore the premium chapter, those have a lock icon in their anchor.
+			local firstPremChapter = v:selectFirst("i.i-lock")
+			if firstPremChapter ~= nil then return end
+			
+			local a = v:selectFirst("a")
+			return NovelChapter {
+				order = i,
+				title = a:child(0):text(),
+				link = shrinkURL(a:attr("href")),
+				release = a:child(1):text()
+			}
+		end))
+
+		info:setChapters(AsList(chapters))
+	end
+	return info
+end
+
+local function parseListing(listingURL)
+	local doc = GETDocument(listingURL)
+	return map(doc:select("li.group"), function(v)
+		local a = v:selectFirst("a")
+		return Novel {
+			title = a:attr("title"),
+			link = shrinkURL(a:attr("href")),
+			imageURL = "https:" .. v:selectFirst("img"):attr("data-src")
+		}
+	end)
+end
+
+local function getSearch(data)
+	local query = data[QUERY]
+	-- Remove the +1 work around when the indexing gets changed.
+	local page = data[PAGE] + 1
+	local status = data[STATUS_FILTER]
+	local orderBy = data[ORDER_BY_FILTER]
+
+	local url = "/all-series/page/" ..page.. "/?"
+	if query ~= nil then
+		url = url .. "search=" .. encode(query)
+	end
+	if status ~= nil then
+		url = url .. "&status=" .. STATUS_VALUES[status+1]:lower()
+	end
+	map(TYPE_VALUES, function(v, i)
+		local KEY_ID = TYPE_FILTER + i
+		if data[KEY_ID] then
+			url = url .. "&type%5B%5D=" .. v:lower():gsub(" ", "-")
+		end
+	end)
+	map(LANGUAGE_VALUES, function(v, i)
+		local KEY_ID = LANGUAGE_FILTER + i
+		if data[KEY_ID] then
+			url = url .. "&language%5B%5D=" .. v:lower():gsub(" ", "-")
+		end
+	end)
+	map(GENRE_VALUES, function(v, i)
+		local KEY_ID = GENRE_FILTER + i
+		if data[KEY_ID] then
+			url = url .. "&genre%5B%5D=" .. v:lower():gsub(" ", "-")
+		end
+	end)
+	if orderBy ~= nil then
+		url = url .. "&orderby=" .. ORDER_BY_TERMS[orderBy+1]
+	end
+	return parseListing(expandURL(url))
```

Comment:
Avoid excessive concatenation, use the `url` library's `querystring`.
See [this review](https://github.com/shosetsuorg/extensions/pull/212#discussion_r774024607) for an example of how to convert your code.

Reply:
Honestly I followed what was done in the Madara Lib, And not sure what makes this a bad practice to a new approach that i have to spend more time into understand its none existent documentation.

I am not going to spend more time on this, I am sorry. I already have the extension working for my own use. if it's unacceptable, do close it.

Reply:
I guess Madara needs fixing too then. I'll ignore it for now.